Expensify/App
# Update the Apple app association file

### Details


Apple is deprecating the format of app-association-file we are currently using, therefore we need to change it. The new format is described in [this documentation](https://developer.apple.com/documentation/xcode/supporting-associated-domains) and the `applinks` property is described more specifically on [this page](https://developer.apple.com/documentation/bundleresources/applinks). 

The file also needs to be served from `.well-known/` directory as described in the docs: 
![image](https://user-images.githubusercontent.com/36083550/142768217-696a0553-2179-4ee4-a515-ffc86ddaef33.png)


Check the linked issue for more details.

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/185770

### Tests


No local tests.

### QA Steps


Once deployed to staging, make sure to have the latest app version and then click [this link](https://staging.new.expensify.com/settings) and make sure the app opens instead of the browser.

Also, put `staging.new.expensify.com` into this Apple tool https://search.developer.apple.com/appsearch-validation-tool and make sure there are no errors.

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
